Bookcase/Patches/StardewValley/Tools/FishingRod/FishCaughtPatch.cs

```diff
@@ -0,0 +1,24 @@
+﻿using Bookcase.Events;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Bookcase.Patches
+{
+    public class FishCaughtPatch : IGamePatch
+    {
+        public Type TargetType => typeof(StardewValley.Tools.FishingRod);
+
+        public MethodInfo TargetMethod => TargetType.GetMethod("pullFishFromWater");
+
+        public static bool Prefix(int whichFish, int fishSize, int fishQuality, int fishDifficulty, bool treasureCaught, bool wasPerfect = false)
+        {
+            AfterFishCaughtEvent caughtEvent = new AfterFishCaughtEvent(whichFish, fishSize, fishQuality, fishDifficulty, treasureCaught, wasPerfect);
+            BookcaseEvents.AfterFishCaught.Post(caughtEvent);
+            return true;
```

Comment:
If this patch can't be canceled, you should set return type to void. bool is optional.